src/pages/PassportRedemption/SignIn/emailScreen.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+
+import { ErrorMessage, SubTitle } from '../style';
+import { Label, Column, InputField } from './index';
+import { ModalPaymentConstants } from '../../../utilities/hooks/ModalPaymentContext';
+import {
+  getPassportEmailId,
+  createPassportEmailId,
+  getPassportTickets,
+} from '../../../utilities/api/interactionManager';
+
+const EmailScreen = () => {
+  const { t } = useTranslation();
+  const { push } = useHistory();
+
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+
+  const viewPassportScreen = async (email) => {
+    const { data } = await getPassportEmailId(email);
+
+    if (!data) {
+      setEmailError(t('passport.errors.noTickets'));
+      setEmail('');
+      return;
+    } else {
+      const { data: hasTickets } = await getPassportTickets(data.id);
+      if (hasTickets.length) push(`/lny-passport/${data.id}/tickets`);
+      else setEmailError(t('passport.errors.noTickets'));
+      return;
+    }
+  };
+
+  const viewUploadScreen = async (email) => {
+    let { data } = await getPassportEmailId(email);
+
+    if (!data) {
+      const { data: newUser } = await createPassportEmailId(email);
+      data = newUser;
+    }
+    push(`/lny-passport/${data.id}/upload`);
```

Comment:
so for me, im not sure why but sometimes the pushing of a new url into history would not always change the screen ,and i realize you would have to use setCurrentScreenView() or however you are changing your screen. I would double check if it both changes the screen and the url just incase/

Reply:
yep! so this is controlled in the index file (the one i showed earlier) so there's both changing of routes + screen changing